### PR TITLE
Default `SIGN_IMAGES` variable to `false`

### PR DESCRIPTION
**Pull request by pierDipi**
nightly release failed with:
```
./hack/../vendor/knative.dev/hack/release.sh: line 314: SIGN_IMAGES: unbound variable
```

Signed-off-by: Pierangelo Di Pilato <pierdipi@redhat.com>

### CODE REVIEW

File: release.sh [hunk 1]
```diff
@@ -311,7 +312,7 @@ function build_from_source() {
 
 # Build a release from source.
 function sign_release() {
-  if [ -z "$SIGN_IMAGES" ]; then # Temporary Feature Gate
+  if [ "$SIGN_IMAGES" ]; then # Temporary Feature Gate
```
Comment (cardil) [inline release.sh:315]:
```suggestion
  if [ "${SIGN_IMAGES:-}" ]; then # Temporary Feature Gate
```

Comment (cardil) [reply release.sh:315]: Then the export above isn't required.

Comment (cardil) [reply release.sh:315]:
```bash
$ SIGN_IMAGES=
$ [ -z $SIGN_IMAGES ]
$ echo $?
0
$ SIGN_IMAGES=1
$ [ -z $SIGN_IMAGES ]
$ echo $?
1
```

Comment (pierDipi) [reply release.sh]: Updated

Comment (pierDipi) [reply release.sh]: thanks!

**Review by cardil — APPROVED** [review]
/lgtm
/approve

**Pull request merged**